strapi/strapi
# Fix: app crashes on upload corrupt image

###  What does it do?

Prevents app from crashing when uploading a corrupt image.

-----
### Why is it needed?

When trying to upload a [corrupt image](https://user-images.githubusercontent.com/64436119/163405278-26d062f5-a695-452d-b88b-01db0d26fa00.png), the application crashes completely.

This can potentially be a high vulnerability issue for any Strapi app if its front end requires the ability to upload images.

-----
### Technical context
When uploading a corrupt image, the application crashes at [optimize](https://github.com/strapi/strapi/blob/master/packages/core/upload/server/services/image-manipulation.js#L88). 

The image is resized and auto-rotated, but for that streaming transformation. There is no error handling. So… 💥  

You might think the error would be a normal throw and be handled elsewhere. But, it seems it doesn’t.

For streams, you have to use the .on('error', () => {})

-----
### How to test it?
Upload this [corrupt image](https://user-images.githubusercontent.com/64436119/163405278-26d062f5-a695-452d-b88b-01db0d26fa00.png) in the Media library.

You should see your app crashing and the website go blank.

Using this fix, the app won't crash, and you will see a new error message below the image 'File is not a supported image'.
NOTE: If anyone has any other idea to improve  the error wording I would appreciate it!

[Reproduce example](https://scribehow.com/shared/Strapi_crashes_when_attempting_to_upload_faulty__unreadable_image_13132__W-GQIQKsTq61Bub2sFhTUQ)


-----
### Related issue(s)/PR(s)
https://github.com/strapi/strapi/issues/13132
